### ISSUE TITLE
API Unification: Add basemap_* methods with unified return structure

### ISSUE THREAD

**Issue by takaeda** [toorpia/toorpia#23]
## Problem Statement

The current toorPIA API has inconsistent return value structures between different methods:

- `fit_transform()` returns `numpy.ndarray` (coordinate data only)
- `fit_transform_csvform()` returns `numpy.ndarray` (coordinate data only) 
- `fit_transform_waveform()` returns `numpy.ndarray` (coordinate data only)
- `addplot_*()` methods return `dict` with structured metadata (`xyData`, `addPlotNo`, `abnormalityStatus`, etc.)

This inconsistency creates confusion and makes the API harder to use, especially when users expect similar functionality from related methods.

### 1. Add New Unified API Methods
- **`basemap_csv()`**: Creates base map from CSV files with unified dict return structure
- **`basemap_waveform()`**: Creates base map from audio/waveform files with unified dict return structure

### 2. Unified Return Structure
All new `basemap_*()` methods will return a consistent dictionary structure:
```python
{
    'xyData': numpy.ndarray,     # Coordinate data (n_samples, 2)
    'mapNo': int,                # Map identification number  
    'shareUrl': str              # Interactive visualization URL
}
```

### 3. API Hierarchy Restructuring
- **Primary APIs**: `basemap_*()` and `addplot_*()` methods (main workflow)
- **Alternative APIs**: `fit_transform()` method (sklearn-compatible, DataFrame processing)

### 4. Backward Compatibility
- Existing `fit_transform_*form()` methods will remain functional
- Add deprecation warnings directing users to new unified APIs
- Maintain all existing functionality and parameter compatibility

### Backend Changes
- New endpoints: `/data/basemap_csv` and `/data/basemap_waveform`
- Reuse existing service logic with unified response formatting

### Client Changes  
- Add `basemap_csv()` and `basemap_waveform()` methods
- Implement unified return structure
- Add deprecation warnings to existing methods

### Documentation Changes
- Restructure README.md with basemap_*/addplot_* as primary workflow
- Update API reference with new method hierarchy
- Add compatibility matrix for method combinations
- Update sample files to demonstrate new APIs

## Benefits

1. **Consistent API Experience**: All related methods return similar structured data
2. **Better Usability**: Users get map metadata (mapNo, shareUrl) along with coordinate data
3. **Clear Workflow**: basemap_* → addplot_* workflow becomes the primary paradigm
4. **Maintained Compatibility**: Existing code continues to work with deprecation guidance

## Testing Plan

- [ ] Verify new basemap_* methods return correct unified structure
- [ ] Confirm backward compatibility of existing methods
- [ ] Validate deprecation warnings function properly
- [ ] Test sample files with new APIs
- [ ] Ensure documentation accuracy

This change improves the overall API consistency while maintaining full backward compatibility.
